django_cte/query.py

```diff
@@ -75,9 +78,22 @@ def generate_sql(cls, connection, query, as_sql):
         for cte in query._with_ctes:
             if django.VERSION > (4, 2):
                 _ignore_with_col_aliases(cte.query)
-            compiler = cte.query.get_compiler(connection=connection)
+
+            elide_empty = True
+            alias = query.alias_map.get(cte.name)
+            if isinstance(alias, QJoin) and alias.join_type == LOUTER:
+                elide_empty = False
```

Comment:
Why is this check needed? Would it work to always pass `elide_empty=False` in this context? Is there a disadvantage to doing that?

Reply:
I'm kind of going with the flow of what Django does as default behavior in the core compiler. The default behavior in the compiler is to raise an assertion error and not run the query in the case of a knowable empty result set. The alternate flow, with `elide_empty` set to false, is used when queries need to be run regardless of the fact that they will not return results.